scripts/quests/otherAreas/The_Gift.lua

```diff
@@ -0,0 +1,93 @@
+-----------------------------------
+-- The Gift
+-----------------------------------
+-- Log ID: 4, Quest ID: 21
+-- Oswald  : !pos 47.119 -15.273 7.989 248
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/quests')
+require('scripts/globals/status')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.THE_GIFT)
+
+quest.reward =
+{
+    item = xi.items.SLEEP_DAGGER,
+    title = xi.title.SAVIOR_OF_LOVE
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.UNDER_THE_SEA) == QUEST_COMPLETED and
+            player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.THE_SAND_CHARM) >= QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SELBINA] =
+        {
+            ['Oswald'] =
```

Comment:
can make this one line ['Oswald'] = quest:progressEvent(70, xi.items.DANCESHROOM), and anywhere else where an NPC has only one dialogue in a portion of the quest (available, accepted, complete)